src/array/mod.rs

```diff
@@ -113,8 +113,13 @@ pub trait MutableArray: std::fmt::Debug {
     /// The optional validity of the array.
     fn validity(&self) -> &Option<MutableBitmap>;
 
-    /// Convert itself to an (immutable) [`Array`].
-    fn as_arc(&mut self) -> Arc<dyn Array>;
+    /// Convert itself to an (immutable) ['Array'].
+    fn as_box(&mut self) -> Box<dyn Array>;
+
+    /// Convert itself to an (immutable) atomically reference counted [`Array`].
+    fn as_arc(&mut self) -> Arc<dyn Array> {
+        self.as_box().into()
```

Comment:
Just a note: this has an extra allocation (`Box` and `Arc` are incompatible), I do not know if rust optimizes this if we inline `as_box` or not. imo this is not a concern, but still a good to know.

Reply:
Looks like possibly not: https://rust.godbolt.org/z/4TrEqGjaf. Maybe it's worth quickly going through and adding explicit impls?

Reply:
As you prefer :)

Reply:
I've added explicit impls for the existing arrays and added a comment to the provided method in the trait.